### PR TITLE
Clarifying documentation of `--static-binary`

### PR DESCRIPTION
**Pull request by yoni206**
The documentation regarding `--static-binary` is improved.

### CODE REVIEW

File: configure.sh [hunk 1]
```diff
@@ -29,7 +29,8 @@ General options;
 Features:
 The following flags enable optional features (disable with --no-<option name>).
   --static                 build static libraries and binaries [default=no]
-  --static-binary          enable/disable static binaries
+  --static-binary          force using standard libraries statically
```
Comment (mpreiner) [inline configure.sh:32]:
```suggestion
  --static-binary          statically link against system libraries
```

File: configure.sh [hunk 2]
```diff
@@ -29,7 +29,8 @@ General options;
...
+                           (must be disabled for static macOS builds) [default=yes]
```
Comment (mpreiner) [inline configure.sh:33]: I wonder if we should do some more magic in the configure.sh script to automatically macOS and disable --static-binary.

Comment (yoni206) [reply configure.sh:33]: Sounds good. If we do that -- can/should we remove `static-binary` option altogether, making it transparent to the user?

Comment (mpreiner) [reply configure.sh:33]: No I'd still keep the option. But since for macOS this can't be supported we can automatically disable it on macOS.

**Review by mpreiner — APPROVED** [review]

**Pull request merged**